overlord/snapstate/snapmgr.go

```diff
@@ -434,11 +441,26 @@ func (m *SnapManager) ensureRefreshes() error {
 	}
 
 	// store attempts in memory so that we can backoff a
-	m.lastRefreshAttempt = time.Now()
-	updated, tasksets, err := AutoRefresh(m.state)
-	if err != nil {
-		return err
-	}
+	delta := timeutil.Next(refreshSchedule, lastRefresh)
+	m.nextRefresh = time.AfterFunc(delta, func() {
+		m.lastRefreshAttempt = time.Now()
+		m.state.Lock()
+		updated, tasksets, err := AutoRefresh(m.state)
+		m.state.Unlock()
+		if err != nil {
+			// FIXME: log error?
+			return
+		}
+		m.doAutoRefresh(updated, tasksets)
+	})
+	logger.Debugf("Next refresh scheduled for %s.", time.Now().Add(delta))
+
+	return nil
+}
+
+func (m *SnapManager) doAutoRefresh(updated []string, tasksets []*state.TaskSet) error {
```

Comment:
this is a strange name given that it's not a task handler, also a bit unclear why the code is split this way between the the function passed to AfterFunc and this one